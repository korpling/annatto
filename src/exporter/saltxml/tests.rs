use graphannis::{update::GraphUpdate, AnnotationGraph};
use insta::assert_snapshot;
use tempfile::TempDir;

use super::*;
use crate::{util::example_generator, StepID};

#[test]
fn export_corpus_structure() {
    let mut updates = GraphUpdate::new();
    example_generator::create_corpus_structure_two_documents(&mut updates);

    // Add some meta data to the documents
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc1".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "description".to_string(),
            anno_value: "A demo corpus".to_string(),
        })
        .unwrap();
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc1".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "author".to_string(),
            anno_value: "unknown".to_string(),
        })
        .unwrap();
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc2".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "description".to_string(),
            anno_value: "Another demo corpus".to_string(),
        })
        .unwrap();

    let mut g = AnnotationGraph::with_default_graphstorages(true).unwrap();
    g.apply_update(&mut updates, |_msg| {}).unwrap();

    let exporter = ExportSaltXml {};
    let output_path = TempDir::new().unwrap();
    let corpus_dir = output_path.path().join("root");
    std::fs::create_dir(&corpus_dir).unwrap();

    let step_id = StepID {
        module_name: "export_saltxml".to_string(),
        path: Some(corpus_dir.clone()),
    };

    exporter
        .export_corpus(&g, &corpus_dir, step_id.clone(), None)
        .unwrap();

    // There should be a saltProject.salt file
    let p = corpus_dir.join("saltProject.salt");
    assert_eq!(true, p.is_file());
    let result = std::fs::read_to_string(p).unwrap();
    assert_snapshot!(result);
}
