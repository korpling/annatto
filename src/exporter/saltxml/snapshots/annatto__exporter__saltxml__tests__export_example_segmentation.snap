---
source: src/exporter/saltxml/tests.rs
expression: doc
---
<?xml version="1.0" encoding="UTF-8"?>
<sDocumentStructure:SDocumentGraph xmlns:sDocumentStructure="sDocumentStructure" xmlns:saltCore="saltCore" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:version="2.0">
  <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1"/>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#b4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::b4"/>
    <labels xsi:type="saltCore:SAnnotation" name="b" value="T::text"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#b3"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::b3"/>
    <labels xsi:type="saltCore:SAnnotation" name="b" value="T::example"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#b2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::b2"/>
    <labels xsi:type="saltCore:SAnnotation" name="b" value="T::other"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#b1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::b1"/>
    <labels xsi:type="saltCore:SAnnotation" name="b" value="T::An"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#a4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::a4"/>
    <labels xsi:type="saltCore:SAnnotation" name="a" value="T::text"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#a3"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::a3"/>
    <labels xsi:type="saltCore:SAnnotation" name="a" value="T::ample"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#a2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::a2"/>
    <labels xsi:type="saltCore:SAnnotation" name="a" value="T::ex"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#a1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::a1"/>
    <labels xsi:type="saltCore:SAnnotation" name="a" value="T::Another"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tli5"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tli5"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tli4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tli4"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tli3"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tli3"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tli2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tli2"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tli1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tli1"/>
  </nodes>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.1" target="//@nodes.0" layers="//@layers.1"/>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.2" target="//@nodes.1" layers="//@layers.1"/>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.3" target="//@nodes.2" layers="//@layers.1"/>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.5" target="//@nodes.4" layers="//@layers.1"/>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.6" target="//@nodes.5" layers="//@layers.1"/>
  <edges xsi:type="sDocumentStructure:SOrderRelation" source="//@nodes.7" target="//@nodes.6" layers="//@layers.1"/>
  <nodes xsi:type="sDocumentStructure:STimeline">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#sTimeline1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::sTimeline1"/>
    <labels xsi:type="saltCore:SFeature" namespace="saltCommon" name="SDATA" value="N::5"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:STextualDS">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#a"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::a"/>
    <labels xsi:type="saltCore:SFeature" namespace="saltCommon" name="SDATA" value="T::Anotherexampletext"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:STextualDS">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#b"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::b"/>
    <labels xsi:type="saltCore:SFeature" namespace="saltCommon" name="SDATA" value="T::Anotherexampletext"/>
  </nodes>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.7" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::2"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.6" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::3"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.5" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::3"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::4"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.4" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::5"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.3" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::1"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.2" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::2"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.1" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::4"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STimelineRelation" source="//@nodes.0" target="//@nodes.13">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::5"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.7" target="//@nodes.14">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::7"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.6" target="//@nodes.14">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::7"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::9"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.5" target="//@nodes.14">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::9"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::14"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.4" target="//@nodes.14">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::14"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::18"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.3" target="//@nodes.15">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::2"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.2" target="//@nodes.15">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::7"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.1" target="//@nodes.15">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::7"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::14"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.0" target="//@nodes.15">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::14"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::18"/>
  </edges>
  <layers xsi:type="saltCore:SLayer">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::l0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::annis"/>
  </layers>
  <layers xsi:type="saltCore:SLayer" edges="//@edges.0 //@edges.1 //@edges.2 //@edges.3 //@edges.4 //@edges.5">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::l1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::default_ns"/>
  </layers>
</sDocumentStructure:SDocumentGraph>
